api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -447,6 +447,20 @@ message HttpConnectionManager {
   config.core.v3.HttpProtocolOptions common_http_protocol_options = 35
       [(udpa.annotations.security).configure_for_untrusted_downstream = true];
 
+  // If :ref:`common_http_protocol_options.max_connection_duration
+  // <envoy_v3_api_field_config.core.v3.HttpProtocolOptions.max_connection_duration>` is
+  // set and this is set to true, the ``max_connection_duration`` setting
+  // will not cause Envoy to ever close HTTP1 connections (other settings are unaffected). Instead,
+  // Envoy will start adding the Connection:close response header to all responses and let the
+  // client choose when to close the connection.
+  //
+  // If ``max_connection_duration`` is set and this is set to false, ``max_connection_duration``
+  // will cause Envoy to enter the normal drain sequence for HTTP1 with Envoy eventually closing the
+  // connection (once there are no active streams).
+  //
+  // Has no effect if ``max_connection_duration`` is unset. Defaults to false.
+  bool http1_safe_max_connection_duration = 58;
```

Comment:
should this be a HttpProtocolOption instead of being in top level hcm config?

IIUC this only allows the client to initiate the connection close. Seems only usable if the downstreams are trusted.

Reply:
I don't think it should be in HttpProtocolOption because that proto is also used for upstream connections. This is specific to the HCM.

Yes, you're exactly right about the client needing to close the connection. There's a race condition for HTTP1 clients that's preventing us from using max_connection_duration at all (described in #34356), so this is still strictly an improvement for us.